app/controllers/schedulers_controller.rb

```diff
@@ -23,9 +23,25 @@ def new; end
   action_auth_level :create, :instructor
   def create
     @scheduler = @course.scheduler.new(scheduler_params)
+    # Check if the action file exists and is readable
+    action_path = Rails.root.join(scheduler_params[:action]).to_path
```

Comment:
Can you also check that the action is within the assessment directory? See `Archive.in_dir?` which for example is used in `handout.rb`.